app/models/customer_resource.rb

```diff
@@ -20,12 +20,41 @@ class CustomerResource < RmApiResource
 
   before_request do |name, request|
     if name == :find_by_package
-      request.get_params[:search] = ''
+      filters = request.get_params.delete(:filter) || {}
+
+      unless filters.is_a?(ActionController::Parameters) || filters.is_a?(Hash)
+        raise ActionController::BadRequest, 'Invalid filter parameter'
+      end
+
+      request.get_params[:selection] =
+        if filters[:selected] == 'true'
+          'selected'
+        elsif filters[:selected] == 'false'
+          'notselected'
+        elsif filters[:selected] == 'ebsco'
+          'orderedthroughebsco'
+        else
+          'all'
+        end
+
+      request.get_params[:resourcetype] = filters[:type] || 'all'
+
+      sort = request.get_params.delete(:sort)
+      request.get_params[:orderby] =
+        if sort == 'relevance'
+          'relevance'
+        elsif sort == 'name'
+          'titlename'
+        else
+          request.get_params[:q] ? 'relevance' : 'titlename'
+        end
+
+      request.get_params[:search] = request.get_params[:q] || ''
```

Comment:
`get_params.delete(:q)` will return the `:q` param, so instead of deleting it below, you could delete it here.

``` ruby
request.get_params[:search] = request.get_params.delete(:q) || ''
```

Reply:
@sadatay had the same sentiment. I tried that @wwilsman but it was failing the test. Will give it another shot now and see if was because of something else i was doing.